sycl/test-e2e/KernelAndProgram/kernel-bundle-get-kernel-ids.cpp

```diff
@@ -0,0 +1,54 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// This test checks that get_kernel_ids returns all the kernels defined
+// in the source regardless of whether they are expressed as lambdas, named
+// function objects or free function kernels.
+
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/oneapi/experimental/free_function_traits.hpp>
+#include <sycl/kernel_bundle.hpp>
+
+class FunctionObjectKernel {
+public:
+  void operator()() const {}
+};
+
+class LambdaKernel;
+
+SYCL_EXT_ONEAPI_FUNCTION_PROPERTY(
+    (sycl::ext::oneapi::experimental::nd_range_kernel<1>))
+void FreeFunctionKernel() {}
+
+int main() {
```

Comment:
The function body doesn't return anything while it's return type is `int` :) I'd suggest returning 0/1 depending on whether the test pass or fails.

Reply:
https://eel.is/c++draft/basic.start.main:
> If control flows off the end of the [compound-statement](https://eel.is/c++draft/stmt.block#nt:compound-statement) of main, the effect is equivalent to a return with operand

there is an implicit `return 0;` at the end of `main()`.

Reply:
As Andrei said, main is treated specially. Nevertheless, I did in fact forgot to explicitly add a return statement so I can add it just for the sake of looking more professional